### ISSUE TITLE
Regression: Adding RateLimit to AES causes pods crash

### ISSUE THREAD

**Issue by tolik93** [datawire/ambassador#3348]
**Describe the bug**
In my case I installed ambassador via helm chart in the same namespace as my application. RateLimitService is enabled via values.yaml of the helm chart. If I add RateLimit to the namespace, the ambassador Pods crash when I open the AES Console. The behavior is same if I create RateLimit in the AES console. Adding `AMBASSADOR_SINGLE_NAMESPACE: true
` and `scope.singleNamespace: true` didn't help. Does anyone have an idea how to fix it quickly?

**To Reproduce**
1. Go to AES Console
2. Click on "Rate Limits"
3. Add sample RateLimit in the custom namespace(not "ambassador") 
4. Ambassador pods will be crash

**Expected behavior**
Creating RateLimit shouldn't cause any crashes of pods. 

**Versions (please complete the following information):**
 - Ambassador: docker.io/datawire/aes:1.12.3
 - Helm chart:   6.6.3
 - Kubernetes environment: AWS EKS
 - Version: 1.19

```yaml
apiVersion: getambassador.io/v2
kind: RateLimit
metadata:
  name: core
spec:
  domain: ambassador
  limits:
  - pattern:
    - generic_key: core
    rate: 500
    unit: minute
```

**Comment by andrei-r-juniper** [comment]
Same problem.  I got partial error from pod log, which shows stack overflow:

```
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0xc022f72398 stack=[0xc022f72000, 0xc042f72000]
fatal error: stack overflow

runtime stack:
runtime.throw(0x2bdecd6, 0xe)
	runtime/panic.go:1116 +0x72
runtime.newstack()
	runtime/stack.go:1060 +0x78d
runtime.morestack()
	runtime/asm_amd64.s:449 +0x8f

goroutine 2422 [running]:
sync.(*poolChain).popTail(0xc000481890, 0x0, 0x0, 0x0)
	sync/poolqueue.go:271 +0x105 fp=0xc022f723a8 sp=0xc022f723a0 pc=0xf33a05
sync.(*Pool).getSlow(0x3d65560, 0x0, 0x0, 0x0)
	sync/pool.go:160 +0x97 fp=0xc022f72400 sp=0xc022f723a8 pc=0xf32c77
sync.(*Pool).Get(0x3d65560, 0x0, 0x0)
	sync/pool.go:137 +0xf6 fp=0xc022f72448 sp=0xc022f72400 pc=0xf32bb6
encoding/json.newEncodeState(0x0)
	encoding/json/encode.go:303 +0x31 fp=0xc022f72478 sp=0xc022f72448 pc=0xfe1451
encoding/json.Marshal(0x299c6e0, 0x3d10960, 0xec2045, 0x28fc680, 0x299c6e0, 0x28c3c01, 0x7fb603a09f20)
	encoding/json/encode.go:159 +0x26 fp=0xc022f724f0 sp=0xc022f72478 pc=0xfe0aa6
github.com/datawire/apro/apis/getambassador.io/v1beta2.RateLimitAction.MarshalJSON(...)
	github.com/datawire/apro/apis/getambassador.io/v1beta2/ratelimit.go:158
github.com/datawire/apro/apis/getambassador.io/v1beta2.(*RateLimitAction).MarshalJSON(0x3d10960, 0x299c6e0, 0x3d10960, 0x7fb603a09f20, 0x3d10960, 0x3d10901)
	<autogenerated>:1 +0x4c fp=0xc022f72538 sp=0xc022f724f0 pc=0x1f31c4c
encoding/json.marshalerEncoder(0xc0106fb480, 0x299c6e0, 0x3d10960, 0x99, 0x100)
	encoding/json/encode.go:477 +0xad fp=0xc022f725b0 sp=0xc022f72538 pc=0xfe23cd
encoding/json.condAddrEncoder.encode(0x2cb6ab0, 0x2cb6b08, 0xc0106fb480, 0x299c6e0, 0x3d10960, 0x99, 0x2990100)
	encoding/json/encode.go:932 +0xb2 fp=0xc022f725e8 sp=0xc022f725b0 pc=0xfe5e92
encoding/json.condAddrEncoder.encode-fm(0xc0106fb480, 0x299c6e0, 0x3d10960, 0x99, 0x2ae0100)
	encoding/json/encode.go:928 +0x6d fp=0xc022f72630 sp=0xc022f725e8 pc=0xff108d
encoding/json.(*encodeState).reflectValue(0xc0106fb480, 0x299c6e0, 0x3d10960, 0x99, 0xc022f70100)
	encoding/json/encode.go:358 +0x82 fp=0xc022f72668 sp=0xc022f72630 pc=0xfe1a62
encoding/json.(*encodeState).marshal(0xc0106fb480, 0x299c6e0, 0x3d10960, 0x2b90100, 0x0, 0x0)
	encoding/json/encode.go:330 +0xf4 fp=0xc022f726c8 sp=0xc022f72668 pc=0xfe1654
encoding/json.Marshal(0x299c6e0, 0x3d10960, 0xec2045, 0x28fc680, 0x299c6e0, 0x28c3c01, 0x7fb603a09f20)
	encoding/json/encode.go:161 +0x52 fp=0xc022f72740 sp=0xc022f726c8 pc=0xfe0ad2
github.com/datawire/apro/apis/getambassador.io/v1beta2.RateLimitAction.MarshalJSON(...)
	github.com/datawire/apro/apis/getambassador.io/v1beta2/ratelimit.go:158
github.com/datawire/apro/apis/getambassador.io/v1beta2.(*RateLimitAction).MarshalJSON(0x3d10960, 0x299c6e0, 0x3d10960, 0x7fb603a09f20, 0x3d10960, 0x3d10901)
	<autogenerated>:1 +0x4c fp=0xc022f72788 sp=0xc022f72740 pc=0x1f31c4c
encoding/json.marshalerEncoder(0xc0106fb400, 0x299c6e0, 0x3d10960, 0x99, 0x100)
	encoding/json/encode.go:477 +0xad fp=0xc022f72800 sp=0xc022f72788 pc=0xfe23cd
encoding/json.condAddrEncoder.encode(0x2cb6ab0, 0x2cb6b08, 0xc0106fb400, 0x299c6e0, 0x3d10960, 0x99, 0x2990100)
	encoding/json/encode.go:932 +0xb2 fp=0xc022f72838 sp=0xc022f72800 pc=0xfe5e92
encoding/json.condAddrEncoder.encode-fm(0xc0106fb400, 0x299c6e0, 0x3d10960, 0x99, 0x2ae0100)
	encoding/json/encode.go:928 +0x6d fp=0xc022f72880 sp=0xc022f72838 pc=0xff108d
encoding/json.(*encodeState).reflectValue(0xc0106fb400, 0x299c6e0, 0x3d10960, 0x99, 0xc022f70100)
	encoding/json/encode.go:358 +0x82 fp=0xc022f728b8 sp=0xc022f72880 pc=0xfe1a62
encoding/json.(*encodeState).marshal(0xc0106fb400, 0x299c6e0, 0x3d10960, 0x2b90100, 0x0, 0x0)
	encoding/json/encode.go:330 +0xf4 fp=0xc022f72918 sp=0xc022f728b8 pc=0xfe1654
...
```

**Comment by esmet** [comment]
Thanks for reporting this! I was able to reproduce the issue from your report and found an issue with RateLimit object JSON marshaling. The fix also appears to be straightforward.

**Comment by khussey** [comment]
This is fixed in the 1.13.3 release, which is now available.

**Comment by khussey** [comment]
See also https://github.com/datawire/apro/issues/2555.
